src/main/java/edu/ie3/datamodel/models/UniqueEntity.java

```diff
@@ -44,4 +44,36 @@ public int hashCode() {
   public String toString() {
     return "UniqueEntity{" + "uuid=" + uuid + '}';
   }
+
+  /**
+   * Abstract class for all builder that build child entities of abstract class {@link UniqueEntity}
+   *
+   * @version 0.1
+   * @since 05.06.20
+   */
+  protected abstract static class UniqueEntityCopyBuilder<T extends UniqueEntityBuilder>
+      implements UniqueEntityBuilder {
+
+    private UUID uuid;
+
+    protected UniqueEntityCopyBuilder(UniqueEntity entity) {
+      this.uuid = entity.getUuid();
+    }
+
+    public T uuid(UUID uuid) {
+      this.uuid = uuid;
+      return childInstance();
+    }
+
+    protected UUID getUuid() {
+      return uuid;
+    }
+
+    protected abstract T childInstance();
```

Comment:
Please add some Documentation on how this pattern / concept works and what the different methods do. Please also consider writing some docs for ReadTheDocs on how to properly use the concept.